deps/process.lua

```diff
@@ -108,13 +108,31 @@ local function removeListener(self, _type, listener)
 end
 
 local function exit(self, code)
+  local completed = 0
+
   code = code or 0
-  local function onStdoutClose()
-    self:emit('exit', code)
-    os.exit(code)
+
+  local function onCompleted()
+    if completed == 2 then
+      self:emit('exit', code)
+      os.exit(code)
+    end
+  end
+
+  local function onStdoutEnd()
+    completed = completed + 1
```

Comment:
You should move the increment into the shared `onCompleted()` function since it's shared among all paths.